airflow/models/xcom.py

```diff
@@ -63,7 +63,7 @@ def init_on_load(self):
         i.e automatically deserialize Xcom value when loading from DB.
         """
         try:
-            self.value = XCom.deserialize_value(self)
+            self.value = XCom.orm_deserialize_value(self)
```

Comment:
why not just change this to `cls.deserialize_value(self)` ? and that way it should use the `deserialize_value` method from Custom backend